example/cajita_tutorial/05_fast_fourier_transform/fast_fourier_transform_example.cpp

```diff
@@ -0,0 +1,208 @@
+/****************************************************************************
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cajita.hpp>
+#include <Cajita_FastFourierTransform.hpp>
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+#include <iostream>
+
+//---------------------------------------------------------------------------//
+// Fast Fourier Transform example.
+//---------------------------------------------------------------------------//
+void FastFourierTransformExample()
+{
+    /*
+      This example shows how to perform a fast fourier transform with Cajita.
+      The current Cajita FFTs take advantage of the heFFTe library, though
+      in the future we plan to support more FFT libraries.
+
+      The basic steps for performing FFTs in Cajita are:
+
+      1. Create a mesh
+
+      2. Create the complex data vector you would like to transform.
+
+      3. Create the FFT and set any options you would like (detailed more below)
+
+*/
+
+    int comm_rank = -1;
+    MPI_Comm_rank( MPI_COMM_WORLD, &comm_rank );
+
+    if ( comm_rank == 0 )
+    {
+        std::cout << "Cajita Fast Fourier Transform Example\n" << std::endl;
+        std::cout << "    (intended to be run with MPI)\n" << std::endl;
+    }
+
+    /*
+       Create the global mesh.
+       In this example we create a uniform 3D periodic mesh, which is not
+       symmetric. Details on creating a global mesh, grid, and partitioner can
+       be found in previous Cajita examples
+
+       Declare the device memory and execution space to use
+       In this example, we use the host space for execution with OpenMP
+    */
+    using MemorySpace = Kokkos::HostSpace;
+    using ExecutionSpace = Kokkos::OpenMP;
```

Comment:
We'll need to only compile with OpenMP with this option - maybe use `DefaultHostExecutionSpace`